### DISCUSSION TITLE
Color not the same between model viewer and quicklook ios

### DISCUSSION THREAD

**Discussion by dungnguyen234** [google/model-viewer#4289] [Q&A]
### Description

the model when display in 3D is totally different color when change to AR quick look ios ,can we edit material of model to fix issue ?
![z4357894958440_517e74fed6850c74503dcc81851ec39c](https://github.com/google/model-viewer/assets/34738412/f3838697-9cf9-4565-854c-e3c9fe9c2580)

![z4357894958331_900f86ececc0ca67c96fc540b5031494](https://github.com/google/model-viewer/assets/34738412/6e98cde3-4566-4c57-8da4-b0e765e63b02)

#### Live Demo

https://glitch.com/edit/#!/model-viewer

### Version

- model-viewer: vX.X.X

### Browser Affected

- [ ] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [x ] Safari

### OS

- [ ] Android
- [ x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [ ] WebXR
- [ ] SceneViewer
- [ ] QuickLook

**Comment by vincentfretin** [comment]
Which version of iOS? iOS 16 changed the default IBL behavior, see https://developer.apple.com/documentation/arkit/usdz_schemas_for_ar/preferrediblversion

**Reply by shahbaziparisa** [reply]
How Can I edit it in modelviewer?

**Reply by elalish** [reply]
This is only possible if Apple added it to their hash param API in Quick Look, but I don't see any indication they did so. @AdaRoseCannon can you help out here with IBL selection? And are either of these IBLs available for download anywhere?

**Comment by dungnguyen234** [comment]
i use ios 16 , any chance to edit IBL behavior in script ?

**Comment by vincentfretin** [comment]
No idea. The models I have looks correct to me on iOS 16+, so I didn't have to do anything.
Also look at the hdr you're using with environment-image attribute on model-viewer, that may also be the main difference.

**Comment by dungnguyen234** [comment]
could you share your model ? so i can check out which is different ?

**Comment by vincentfretin** [comment]
Literally the default cube colored in red exported from blender in glb, with the auto generated usdz from glb I have the same red in model-viewer without environment-image (so neutral) and Quicklook Object mode.

**Comment by vincentfretin** [comment]
Are you using a usdz you created or the usdz from glb that is generated automatically? Are you using a texture? linear or sRGB color space?

**Comment by dungnguyen234** [comment]
i use usdz from glb auto generated  only

**Comment by dungnguyen234** [comment]
> Are you using a usdz you created or the usdz from glb that is generated automatically? Are you using a texture? linear or sRGB color space?

This model is auto generated from GLB and texture is sRGB

**Comment by vincentfretin** [comment]
If you want to compare, look at the models in https://modelviewer.dev/examples/augmentedreality/index.html
like the train or the canoe, those examples are using the neutral environment.
I don't have any color difference on my iPad iOS 16.4.1.

**Comment by elalish** [comment]
Great discussion, and thanks for the IBL pointer @vincentfretin! Rendering consistency between glTF and USDZ has been a big problem because Apple hasn't spec'ed out USDZ well enough for us to know how they render. Anything we determine here will be very useful to our other users, so please comment with any findings, particularly more screenshots, so we can see what settings work best.

**Comment by aarivtmk** [comment]
Hey @elalish 
Even I am facing the same issue
so I am simply fetching glb file from s3 and creating a theme app extension app for Shopify with React.

here its shows fine
<img width="398" alt="Screenshot 2024-05-26 at 07 58 12" src="https://github.com/google/model-viewer/assets/62956460/9dbe8cb2-b9ee-43d0-8208-707c36c44078">

but when I switch to AR mode, the color is becoming dark on the material
![img_1649_720](https://github.com/google/model-viewer/assets/62956460/45f0514a-1c80-414c-a45a-b57957814aee)

![img_1648_720](https://github.com/google/model-viewer/assets/62956460/df7f2ea1-4a56-44b7-96c5-86618bd4ea34)

I thought changing the tone mapping would work, but so far, no luck. could you please take a look?

Thanks in advance

**Reply by elalish** [reply]
Is that last picture webxr (android, in browser) mode? That should be the same renderer as on the page, so the difference there would have to be lighting. Are you doing anything fancy? Can you include your `<model-viewer>` HTML line here?

**Reply by aarivtmk** [reply]
Hey @elalish 
on android, there are no issues
and the above pictures are from ios device

```html
<model-viewer
        id="model-viewer"
        className="glb_view mx-auto d-block"
        double-sided
        ar-modes="webxr scene-viewer quick-look"
        camera-controls
        touch-action="pan-y"
        disable-tap
        key={props.modelViewerKey}
        style={{ height: viewerHeight, width: "100%" }}
        alt="GLB model"
        src={props.glbFile}
        ar
        loading="eager"
        shadow-intensity="1"
        xr-environment
      >
        {/* <div
          id="mobile-ar-text"
          className="text-right fw-bold d-none d-md-block"
        >
          Use Mobile AR to see in your room
        </div> */}
        <button
          slot="ar-button"
          id="ar-button"
          className="mx-auto col-5"
          style={{
            position: "absolute",
            bottom: "0px",
            right: "0px",
            zIndex: 10,
          }}
        >
          See in your room
          <svg
            version="1.1"
            id="view_in_AR_icon"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
            viewBox="0 0 24 24"
            style={{ enableBackground: "new 0 0 24 24" }}
            xmlSpace="preserve"
          >
            <rect id="Bounding_Box" fill="none" width="24" height="24" />
            <g id="Art_layer">
              <path d="M3,4c0-0.55,0.45-1,1-1h2V1H4C2.35,1,1,2.35,1,4v2h2V4z" />
              <path d="M20,3c0.55,0,1,0.45,1,1v2h2V4c0-1.65-1.35-3-3-3h-2v2H20z" />
              <path d="M4,21c-0.55,0-1-0.45-1-1v-2H1v2c0,1.65,1.35,3,3,3h2v-2H4z" />
              <path d="M20,21c0.55,0,1-0.45,1-1v-2h2v2c0,1.65-1.35,3-3,3h-2v-2H20z" />
              <g>
                <path d="M18.25,7.6l-5.5-3.18c-0.46-0.27-1.04-0.27-1.5,0L5.75,7.6C5.29,7.87,5,8.36,5,8.9v6.35c0,0.54,0.29,1.03,0.75,1.3 l5.5,3.18c0.46,0.27,1.04,0.27,1.5,0l5.5-3.18c0.46-0.27,0.75-0.76,0.75-1.3V8.9C19,8.36,18.71,7.87,18.25,7.6z M7,14.96v-4.62 l4,2.32v4.61L7,14.96z M12,10.93L8,8.61l4-2.31l4,2.31L12,10.93z M13,17.27v-4.61l4-2.32v4.62L13,17.27z" />
              </g>
            </g>
          </svg>
        </button>
      </model-viewer>
```

**Reply by aarivtmk** [reply]
Hey @elalish 
Could you please take a look?

**Reply by elalish** [reply]
Alas, this is due to differences in iOS lighting estimation and tone mapping, over which we have no control. @AdaRoseCannon is Apple interested in trying to get consistent rendering between devices? Because if so, we should talk.

**Reply by shahbaziparisa** [reply]
Hi. I have the same problem. Did you find any good way?
